compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -1015,7 +1018,13 @@ fn parse_type_inner(
         named_type(recursive_type_parser.clone()),
         named_trait(recursive_type_parser.clone()),
         array_type(recursive_type_parser.clone()),
-        recursive_type_parser.clone().delimited_by(just(Token::LeftParen), just(Token::RightParen)),
+        recursive_type_parser
+            .clone()
+            .delimited_by(just(Token::LeftParen), just(Token::RightParen))
+            .map_with_span(|typ, span| UnresolvedType {
+                typ: UnresolvedTypeData::Parenthesized(Box::new(typ)),
+                span: span.into(),
+            }),
```

Comment:
Since this is multiple lines now, can you break it out into a new function, `parenthesized_expression`?